pkgs/by-name/op/openscap/package.nix

```diff
@@ -0,0 +1,158 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  libsepol,
+  popt,
+  libxml2,
+  libxslt,
+  openssl,
+  util-linux,
+  pcre2,
+  pcre,
+  libselinux,
+  graphviz,
+  glib,
+  python3,
+  swig,
+  python3Packages,
+  libgcrypt,
+  opendbx,
+  xmlbird,
+  haskellPackages,
+  libyaml,
+  xmlsec,
+  bzip2,
+  yamlscript,
+  yaml-cpp,
+  valgrind,
+  asciidoc,
+  installShellFiles,
+  rpm,
+  system-sendmail,
+  hyperscan,
+  gnome2,
+  curl,
+  procps,
+  systemd,
+  perl,
+  doxygen,
+  pkg-config,
+  perl538Packages,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "openscap";
+  version = "1.3.10";
+
+  src = fetchFromGitHub {
+    owner = "OpenSCAP";
+    repo = "openscap";
+    rev = version;
+    hash = "sha256-P7k+Ygz/XmpTSKBEqbyJsd1bIDVJ1HA/RJdrEtjYD5g=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    doxygen
+    pkg-config
+  ];
+
+  buildInputs =
+    with perl538Packages;
+    [
+      XMLXPath
+      LinuxACL
+      XMLTokeParser
+    ]
+    ++ [
+      rpm
+      perl
+      popt
+      openssl
+      valgrind
+      pcre2
+      pcre
+      swig
+      libxslt
+      xmlsec
+      hyperscan
+      yaml-cpp
+      util-linux
+      libselinux
+      libyaml
+      xmlbird
+      installShellFiles
+      bzip2
+      python3
+      libgcrypt
+      libxml2
+      systemd
+      yamlscript
+      haskellPackages.pthread
+      graphviz
+      system-sendmail
+      procps
+      asciidoc
+      libsepol
+      curl
+      glib
+      gnome2.ORBit2
+      gnome2.GConf
+      opendbx
+    ];
+
+  prePatch = ''
+    substituteInPlace swig/perl/CMakeLists.txt --replace-fail "DESTINATION ''${PERL_VENDORLIB}" "DESTINATION $out/lib''${PERL_VERSION}"
+    substituteInPlace swig/perl/CMakeLists.txt --replace-fail "DESTINATION ''${PERL_VENDORARCH}" "DESTINATION $out/lib"
+    substituteInPlace src/common/oscap_pcre.c --replace "#include <pcre2.h>" "#include <${pcre2.dev}/include/pcre2.h>"
+    substituteInPlace src/common/oscap_pcre.c --replace "#include <pcre.h>" "#include <${pcre.dev}/include/pcre.h>"
+  '';
+
+  cmakeFlags = [
+    "-DPCRE2_INCLUDE_DIRS=${pcre2.dev}/include"
+    "-DPCRE2_LIBRARIES=${pcre2.out}/lib"
+    "-DENABLE_DOCS=TRUE"
+    "-DENABLE_TESTS=TRUE"
+    "-DENABLE_OSCAP_UTIL=TRUE"
+    "-DENABLE_OSCAP_UTIL_CHROOT=TRUE"
+    "-DENABLE_OSCAP_UTIL_SSH=TRUE"
+    "-DENABLE_OSCAP_UTIL_DOCKER=TRUE"
+    "-DENABLE_VALGRIND=TRUE"
+    "-DPYTHON_SITE_PACKAGES_INSTALL_DIR=lib/${python3Packages.python.interpreter}"
```

Comment:
shouldn't this be `python3.pkgs.python.sitePackages`?

Reply:
you are correct but cmake does't recognise it as an output path unless some output is specified like `lib` otherwise build process fails. to complete the build process an output is needed but it still produces result like $out/lib/ni/store/8o.../python3.12/lib/openscap_pm.pm. any solution to prevent such output?

Reply:
But `lib/${python3Packages.python.interpreter}"`
evaluates to `lib//nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/bin/python3.12` (not lib).
So you would need 
`${python3}/${python3.pkgs.python.sitePackages}` which evaluates to
`/nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/lib/python3.12/site-packages` right?

Reply:
result of the solution: "/nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/lib/python3.12/site-packages/oscap_docker_python":
  Permission denied.

Reply:
ok. But AFAIK CMAKE should install python dependencies in `$out/lib/python3.12/site-packages` so, this should be
`"${python3.pkgs.python.sitePackages}"` which will evaluate to `lib/python3.12/site-packages`

Reply:
solution worked but when testing in local machine i used `${python3.sitePackages}` instead. for the openscap_pm.pm library i made adjustment in prepatch DESTINATION to avoid wrong output. All fine now. I need this #327164 merged before merging openscap. important dependency for buildig yaml path still missing.